skymarshal/dexserver/dexserver.go

```diff
@@ -217,3 +217,32 @@ func newLocalUsers(config *DexConfig) map[string][]byte {
 	return users
 
 }
+
+func newLocalClients(config *DexConfig) map[string][]byte {
+	clients := map[string][]byte{}
+
+	for clientID, clientSecret := range config.Clients {
+		if clientID != "" && clientSecret != "" {
+
+			var hashed []byte
+
+			if _, err := bcrypt.Cost([]byte(clientSecret)); err != nil {
+				if hashed, err = bcrypt.GenerateFromPassword([]byte(clientSecret), 0); err != nil {
+
+					config.Logger.Error("bcrypt-client-secret", err, lager.Data{
+						"client_id": clientID,
+					})
+
+					continue
```

Comment:
I wonder if it makes sense to just log an error here. Admittedly, I'm not sure if/when this error would ever occur in practice, but maybe it would make sense to short circuit instead?

If we fail to create a client [here](https://github.com/concourse/concourse/pull/5459/files#diff-097bd19334894da24754e934c93e4638R159-R161), we return an error - I would expect the same behaviour for failing to generate the hash (since we'd end up without that client in that case).

WDYT? My guess is this error is very unlikely to occur in practice, so this may be a moot point.

Reply:
I honestly didn't give it much thought. I basically copied the same implementation we had for user passwords. I can't remember how we originally settled on logging the error and continuing, it does seem a bit weird, but I think it was for a reason.